zebra-state/src/service/check/anchors.rs

```diff
@@ -55,46 +55,67 @@ pub(crate) fn anchors_refer_to_earlier_treestates(
             // > output treestate of any prior JoinSplit description in the same transaction.
             //
             // https://zips.z.cash/protocol/protocol.pdf#joinsplit
-            let mut interstitial_roots: HashSet<sprout::tree::Root> = HashSet::new();
-
-            let mut interstitial_note_commitment_tree = parent_chain.sprout_note_commitment_tree();
+            let mut interstitial_trees: HashMap<
+                sprout::tree::Root,
+                sprout::tree::NoteCommitmentTree,
+            > = HashMap::new();
 
             for joinsplit in transaction.sprout_groth16_joinsplits() {
                 // Check all anchor sets, including the one for interstitial anchors.
                 //
+                // The anchor is checked and the matching tree is obtained, which
+                // is used to create the interstitial tree state for this JoinSplit:
+                //
+                // > For each JoinSplit description in a transaction, an interstitial output
+                // > treestate is constructed which adds the note commitments and nullifiers
+                // > specified in that JoinSplit description to the input treestate referred
+                // > to by its anchor. This interstitial output treestate is available for use
+                // > as the anchor of subsequent JoinSplit descriptions in the same transaction.
+                //
+                // https://zips.z.cash/protocol/protocol.pdf#joinsplit
+                //
                 // Note that [`interstitial_roots`] is always empty in the first
                 // iteration of the loop. This is because:
                 //
                 // > "The anchor of each JoinSplit description in a transaction
                 // > MUST refer to [...] to the interstitial output treestate of
                 // > any **prior** JoinSplit description in the same transaction."
```

Comment:
I would include the full quote here; it's not too much longer. (The missing part is "either some earlier block's final Sprout treestate, or", which is important.)

Reply:
Thank you @daira. I added that, and I rewrote the docs so that they comply with the new way of documenting the consensus rules. We recently agreed on a new way of documenting the consensus rules to facilitate the upcoming audit.